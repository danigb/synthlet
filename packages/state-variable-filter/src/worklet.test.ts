import { getProcessorName } from "./index";

describe("ProcessorNode", () => {
  let Processor: any;
  const sampleRate = 40;

  beforeAll(async () => {
    createWorkletTestContext(sampleRate);
    Processor = (await import("./worklet")).Processor;
  });

  it("registers processor", () => {
    expect(global.registerProcessor).toHaveBeenCalledWith(
      getProcessorName(),
      Processor
    );
  });

  it("has parameter descriptors", () => {
    expect(Processor.parameterDescriptors).toMatchSnapshot();
  });
});

function createWorkletTestContext(sampleRate = 10) {
  // @ts-ignore
  global.sampleRate = sampleRate;
  // @ts-ignore
  global.AudioWorkletProcessor = class AudioWorkletNodeStub {
    port: {
      postMessage: jest.Mock<any, any, any>;
      onmessage: jest.Mock<any, any, any>;
    };

    constructor() {
      this.port = {
        postMessage: jest.fn(),
        onmessage: jest.fn(),
      };
    }
  };
  // @ts-ignore
  global.registerProcessor = jest.fn(); // Mock registerProcessor
}
