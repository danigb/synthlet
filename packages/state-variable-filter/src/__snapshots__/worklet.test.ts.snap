// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessorNode has parameter descriptors 1`] = `
[
  {
    "automationRate": "k-rate",
    "defaultValue": 1,
    "maxValue": 3,
    "minValue": 0,
    "name": "filterType",
  },
  {
    "automationRate": "k-rate",
    "defaultValue": 1000,
    "maxValue": 20000,
    "minValue": 16,
    "name": "frequency",
  },
  {
    "automationRate": "k-rate",
    "defaultValue": 0.5,
    "maxValue": 40,
    "minValue": 0,
    "name": "resonance",
  },
]
`;
